Sources/StreamChatUI/Composer/ComposerVC.swift

```diff
@@ -893,44 +903,49 @@ open class ComposerVC: _ViewController,
         )
     }
 
+    /// Returns the query to be used for searching members inside a channel.
+    ///
+    /// This function is called in `showMentionSuggestions` to retrieve the query
+    /// that will be used to search for members. You should override this if you want to change the
+    /// member searching logic.
+    ///
+    /// - Parameter typingMention: The potential user mention the current user is typing.
+    /// - Returns: `ChannelMemberListQuery` instance that will be used for searching members in a channel.
+    open func queryForChannelMentionSuggestionsSearch(typingMention term: String) -> ChannelMemberListQuery? {
+        guard let cid = channelController?.cid else {
+            return nil
+        }
+        return ChannelMemberListQuery(
+            cid: cid,
+            filter: .autocomplete(.name, text: term),
+            sort: [.init(key: .name, isAscending: true)]
+        )
+    }
+
+    /// Returns the member list controller to be used for searching members inside a channel.
+    ///
+    /// - Parameter term: The potential user mention the current user is typing.
+    /// - Returns: `ChatChannelMemberListController` instance that will be used for searching members in a channel.
+    open func makeMemberListControllerForMemberSuggestions(typingMention term: String) -> ChatChannelMemberListController? {
```

Comment:
Would it be better to take a ChannelMemberListQuery as an argument here for forcing subclassers to override once function for query and the other for UI? Currently I could just override this function and skip the `queryForChannelMentionSuggestionsSearch` all together. WDYT? Then subclassers have less ways for making a mistake and getting confused (someone overrides makeMember, other dev overrides query, but later it turns out that the makeMember subclasser never calls queryFor anymore, these kinds of simple issues).

Reply:
that's a good point

Reply:
Yeah, I think it makes sense, although it removes some flexibility, in case we have other ways of initialising the controller without a member query 🤔 But in this case it should be unlikely tho

Reply:
Actually, I would prefer to have only `makeMemberListControllerForMemberSuggestions(typingMention:)` and not `queryForChannelMentionSuggestionsSearch`, the only reason I created the `queryForChannelMentionSuggestionsSearch` was because we already had `queryForMentionSuggestionsSearch`, so I wanted to have consistency. But having only `makeMemberListControllerForMemberSuggestions(typingMention:)` would be enough and the most flexible approach. 

The disadvantage of your suggestion is that it will always require doing 2 function calls to pass to create the memberListController, one for creating the member list query and another for creating the actual controller. 

Right now, we just do this:
```swift
memberListController = makeMemberListControllerForMemberSuggestions(typingMention: typingMention)
```

With your suggestion, we would need to do this:
```swift
let query = queryForChannelMentionSuggestionsSearch(typingMention: typingMention)
memberListController = makeMemberListControllerForMemberSuggestions(query: query)
```

The 2nd approach has some disadvantages:
1- If a customer wants to create the member list controller in another place and wants to make sure that it uses the same query as in the other places, they will always need to follow this pattern. They might forget to call `queryForChannelMentionSuggestionsSearch(typingMention: typingMention)` or know that it actually exists, and so they create a MemberListQuery from scratch. 
2- It forces having a MemberListQuery to create the memberListController, which right now is a good thing, but it removes some flexibility which is something that is usually a bad practice in the UI SDK since customers usually have a lot of different use cases.

Reply:
Discussed on Slack ✅